### ISSUE TITLE
az login: TypeError: 'NoneType' object is not iterable

### ISSUE THREAD

**Issue by savvaniss** [Azure/azure-cli#16014]
## Describe the bug

**Command Name**
`az login`

**Errors:**
```
'NoneType' object is not iterable
Traceback (most recent call last):
python3.6/site-packages/knack/cli.py, ln 215, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 654, in execute
    raise ex
cli/core/commands/__init__.py, ln 718, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 711, in _run_job
    six.reraise(*sys.exc_info())
...
cli/command_modules/profile/custom.py, ln 158, in login
    use_cert_sn_issuer=use_cert_sn_issuer)
azure/cli/core/_profile.py, ln 240, in find_subscriptions_on_login
    is_service_principal, bool(use_cert_sn_issuer))
azure/cli/core/_profile.py, ln 276, in _normalize_properties
    subscription_dict[_MANAGED_BY_TENANTS] = [{_TENANT_ID: t.tenant_id} for t in s.managed_by_tenants]
TypeError: 'NoneType' object is not iterable
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az login`

## Environment Summary
```
Linux-4.4.0-19041-Microsoft-x86_64-with-debian-bullseye-sid
Python 3.6.10
Installer: DEB

azure-cli 2.12.1 *

```

**Comment by yonzhan** [comment]
az login

**Comment by watsonb** [comment]
Any updates on this?  I'm suddenly running into this now.

```bash
sh-4.2# az --version
azure-cli                         2.12.0 *

core                              2.12.0 *
telemetry                          1.0.6

Python location '/usr/bin/python3'
Extensions directory '/home/awx/.azure/cliextensions'

Python (Linux) 3.6.8 (default, Nov 16 2020, 16:55:22) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
```

**Comment by jiasli** [comment]
This is because you used an incorrect API version/profile on Azure Stack and triggered an Azure Stack bug https://github.com/Azure/azure-rest-api-specs/issues/9567.

I have already fixed it in https://github.com/Azure/azure-cli/commit/24fe6f6a5843518639f0f3a16a9b7a3e64e103e2 and the change is included since `2.15.0`.
